{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Registered S3 methods overwritten by 'ggplot2':\n",
      "  method         from \n",
      "  [.quosures     rlang\n",
      "  c.quosures     rlang\n",
      "  print.quosures rlang\n",
      "Registered S3 method overwritten by 'rvest':\n",
      "  method            from\n",
      "  read_xml.response xml2\n",
      "── Attaching packages ─────────────────────────────────────── tidyverse 1.2.1 ──\n",
      "✔ ggplot2 3.1.1       ✔ purrr   0.3.2  \n",
      "✔ tibble  2.1.1       ✔ dplyr   0.8.0.1\n",
      "✔ tidyr   0.8.3       ✔ stringr 1.4.0  \n",
      "✔ readr   1.3.1       ✔ forcats 0.4.0  \n",
      "── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "✖ dplyr::filter() masks stats::filter()\n",
      "✖ dplyr::lag()    masks stats::lag()\n",
      "\n",
      "Attaching package: ‘MASS’\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "Attaching package: ‘reshape2’\n",
      "\n",
      "The following object is masked from ‘package:tidyr’:\n",
      "\n",
      "    smiths\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(tidyverse)\n",
    "library(MASS)\n",
    "library(DescTools)\n",
    "library(pwr)\n",
    "library(repr)\n",
    "library(cowplot)\n",
    "library(ggseqlogo)\n",
    "library(reshape2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = c('chromosome', 'reference_nucleotide', 'position', 'trinuc_context','dinuc_context', \n",
    "         \"methylation_frequency\", \"count_mC\", \"coverage\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "#####HELPERS \n",
    "\n",
    "load_cgmap <- function(cgmap,strain, replicate, growth_phase, metabolic_condition='S'){\n",
    "    cols = c('chromosome', 'base', 'position', 'trinuc_context','dinuc_context', \n",
    "         \"methylation_frequency\", \"count_mC\", \"coverage\")\n",
    "    outdata <- read.delim(cgmap, sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "        mutate(strain=strain, growth_phase=growth_phase, \n",
    "               metabolic_condition=metabolic_condition, replicate=replicate)\n",
    "\n",
    "}\n",
    "\n",
    "\n",
    "#clean up raw CGmaps\n",
    "processCGmap <- function(cgmap){\n",
    "    strain = unique(cgmap$strain)\n",
    "    rep = unique(cgmap$replicate)\n",
    "    \n",
    "    final_data <- cgmap %>% \n",
    "        dplyr::select(-trinuc_context, -dinuc_context) %>%\n",
    "        dplyr::rename(!!paste(\"coverage\",\"_\",strain,\"_\",rep, sep=\"\") := coverage) %>%\n",
    "        dplyr::rename(!!paste(\"mC_count\",\"_\",strain,\"_\",rep, sep=\"\") := count_mC) %>%\n",
    "        dplyr::rename(!!paste(\"methylation_frequency\",\"_\",strain,\"_\",rep, sep=\"\") := methylation_frequency) %>%\n",
    "        dplyr::select(-replicate, -strain)\n",
    "    return(final_data)\n",
    "}\n",
    "\n",
    "# get standard deviation for 3 reps\n",
    "get_sd <- function(f1,f2,f3,c1,c2,c3,min_cov){\n",
    "    ls <-c()\n",
    "    if(c1>=min_cov){\n",
    "        ls <- c(ls,f1)\n",
    "    }\n",
    "    if(c2>=min_cov){\n",
    "        ls <- c(ls, f2)\n",
    "\n",
    "    }\n",
    "    \n",
    "    if(c3>=min_cov){\n",
    "       ls <- c(ls, f3)\n",
    "\n",
    "    }\n",
    "    if (length(ls>1)){\n",
    "        ans = round(sd(ls),3)\n",
    "    } else{\n",
    "        ans = NA\n",
    "    }\n",
    "    return(ans)\n",
    "}\n",
    "\n",
    "\n",
    "# get standard deviation for 2 reps\n",
    "get_sd_2reps <- function(f1,f2,c1,c2,min_cov=47){\n",
    "    ls <-c()\n",
    "    if(c1>=min_cov){\n",
    "        ls <- c(ls,f1)\n",
    "    }\n",
    "    if(c2>=min_cov){\n",
    "        ls <- c(ls, f2)\n",
    "    }\n",
    "    if (length(ls>1)){\n",
    "        ans = round(sd(ls),3)\n",
    "    } else{\n",
    "        ans = NA\n",
    "    }\n",
    "    return(ans)\n",
    "}\n",
    "\n",
    "\n",
    "#merge data for replicate deletion strains\n",
    "merge_del_reps <- function(x,y, strain){\n",
    "    cov_1 <- paste(\"coverage_\",strain, \"_Dec2016\", sep='')\n",
    "    cov_2 <- paste(\"coverage_\",strain, \"_Oct2020\", sep='')\n",
    "\n",
    "    d <- x %>%\n",
    "        merge(y=y, \n",
    "              by=c(\"chromosome\",\"position\", 'reference_nucleotide', 'growth_phase','metabolic_condition'), \n",
    "              all=TRUE) %>%\n",
    "    \n",
    "            mutate(!!as.name(cov_1) := replace_na(!!as.name(cov_1),0)) %>%\n",
    "            mutate(!!as.name(cov_2) := replace_na(!!as.name(cov_2),0))\n",
    "    \n",
    "    return(d)\n",
    "}\n",
    "\n",
    "\n",
    "# determine if high confidence, reproducible sites with from 3 reps\n",
    "detect_hiconf_3reps <- function(f1,m1,p1,c1,f2,m2,p2,c2,f3,m3,p3,c3, min_cov=47,\n",
    "                                min_freq=.10, min_freq2 = 0.05) {\n",
    "    ls <- c()\n",
    "    \n",
    "    if(c1 == 'NA' ) {c1<-0}\n",
    "    if(c2 == 'NA' ) {c2<-0}\n",
    "    if(c3 == 'NA' ) {c3<-0}\n",
    "    if(c1 < min_cov & c2 < min_cov & c3 < min_cov) {return(NA)}\n",
    "    \n",
    "    else{\n",
    "    \n",
    "    \n",
    "    if(c1<min_cov){ls <- c(ls,\"NA\")}\n",
    "    else if(  (c1>=min_cov & m1>=p1 & f1>=min_freq) | (m1>50 & f1>=min_freq2) ) {ls <- c(ls,TRUE)}\n",
    "    else {ls <- c(ls,FALSE)}\n",
    "    \n",
    "    if(c2<min_cov){ls <- c(ls,\"NA\")}\n",
    "    else if( (c2>=min_cov & m2>=p2 & f2>=min_freq) | (m2>50 & f2>=min_freq2) ) {ls <- c(ls,TRUE)}\n",
    "    else {ls <- c(ls,FALSE)}\n",
    "    \n",
    "    if(c3<min_cov){ls <- c(ls,\"NA\")}\n",
    "    else if( (c3>=min_cov & m3>=p3 & f3>=min_freq)| (m3>50 & f3>=min_freq2) )  {ls <- c(ls,TRUE)}\n",
    "    else {ls <- c(ls,FALSE)}\n",
    "    \n",
    "    ans_t <- length(which(ls==TRUE))\n",
    "    ans_na <- length(which(ls==\"NA\"))\n",
    "    ans_f <- length(which(ls==FALSE))\n",
    "\n",
    "    if(ans_na >= 3){return(NA)}\n",
    "    if(ans_na == 2 & ans_t == 1){return(NA)}\n",
    "    else if(ans_t >= 2){return(TRUE)}\n",
    "    else { return(FALSE)}\n",
    "        }\n",
    "}\n",
    "\n",
    "\n",
    "\n",
    "# determine if high confidence, reproducible sites with from 2 reps\n",
    "detect_hiconf_2reps <- function(f1,m1,p1,c1,f2,m2,p2,c2, min_cov=47,\n",
    "                                min_freq=.10, min_freq2 = 0.05) {\n",
    "    ls <- c()\n",
    "    \n",
    "    if(c1 == 'NA' ) {c1<-0}\n",
    "    if(c2 == 'NA' ) {c2<-0}\n",
    "    \n",
    "    if(c1 < min_cov & c2 < min_cov) {return(NA)}\n",
    "    else{\n",
    "    \n",
    "    \n",
    "    if(c1<min_cov){ls <- c(ls,\"NA\")}\n",
    "    else if(  (c1>=min_cov & m1>=p1 & f1>=min_freq) | (m1>50 & f1>=min_freq2) ) {ls <- c(ls,TRUE)}\n",
    "    else {ls <- c(ls,FALSE)}\n",
    "    \n",
    "    if(c2<min_cov){ls <- c(ls,\"NA\")}\n",
    "    else if( (c2>=min_cov & m2>=p2 & f2>=min_freq) | (m2>50 & f2>=min_freq2) ) {ls <- c(ls,TRUE)}\n",
    "    else {ls <- c(ls,FALSE)}\n",
    "\n",
    "    ans_t <- length(which(ls==TRUE))\n",
    "    ans_na <- length(which(ls==\"NA\"))\n",
    "    ans_f <- length(which(ls==FALSE))\n",
    "\n",
    "    if(ans_f >= 1  ){return(FALSE)}\n",
    "    else if(ans_t >= 2){return(TRUE)}\n",
    "    else if(ans_t == 1 & ans_na ==1){return(TRUE)}\n",
    "    else { return(FALSE)}\n",
    "        \n",
    "        }\n",
    "}\n",
    "\n",
    "    \n",
    "    \n",
    "# determine if high confidence, reproducible sites with from 1 rep\n",
    "detect_hiconf_1rep <- function(f1,m1,p1,c1, min_cov=47, min_freq=.10,min_freq2=0.05) {\n",
    "   \n",
    "    if(c1 == 'NA') {c1<-0}\n",
    "    \n",
    "    if(c1 < min_cov) {return(NA)}\n",
    "    else if( (c1>=min_cov & m1>=p1 & f1>=min_freq) | (m1>50 & f1>=min_freq2) ) {return(TRUE)}\n",
    "    else {return(FALSE)}\n",
    "    \n",
    "}\n",
    "\n",
    "# determine if high confidence in all three 3 reps\n",
    "super_hiconf_3reps <- function(f1,m1,p1,c1,f2,m2,p2,c2,f3,m3,p3,c3, min_cov=47,\n",
    "                                min_freq=.10, min_freq2 = 0.05) {\n",
    "    ls <- c()\n",
    "    \n",
    "    if(c1 == 'NA' ) {c1<-0}\n",
    "    if(c2 == 'NA' ) {c2<-0}\n",
    "    if(c3 == 'NA' ) {c3<-0}\n",
    "\n",
    "    \n",
    "    if(  (c1>=min_cov & m1>=p1 & f1>=min_freq) | (c1>1000 & f1>=min_freq2)  ) {ls <- c(ls,TRUE)}\n",
    "    else {ls <- c(ls,FALSE)}\n",
    "    \n",
    "    \n",
    "    if( (c2>=min_cov & m2>=p2 & f2>=min_freq) | (c2>=1000 & f2>=min_freq2) ) {ls <- c(ls,TRUE)}\n",
    "    else {ls <- c(ls,FALSE)}\n",
    "    \n",
    "    if( (c3>=min_cov & m3>=p3 & f3>=min_freq) | (c3>=1000 & f3>=min_freq2) )  {ls <- c(ls,TRUE)}\n",
    "    else {ls <- c(ls,FALSE)}\n",
    "    \n",
    "    ans <- length(which(ls==TRUE))\n",
    "    \n",
    "    if(ans >= 3){return(TRUE)}\n",
    "    else { return(FALSE)}\n",
    "}\n",
    "\n",
    "\n",
    "\n",
    "#compare standard deviations\n",
    "rm_overlap <- function(sp, ssd, cp,csd){\n",
    "    ssd <- ssd*2\n",
    "    csd <- csd*2\n",
    "\n",
    "    r1 <- c(sp-ssd, sp+ssd)\n",
    "    r2 <- c(cp-csd, cp+csd)\n",
    "    ans <- r1 %overlaps% r2\n",
    "    return(ans)\n",
    "    \n",
    "}\n",
    "\n",
    "\n",
    "# get p value of each m5C site using binomial distribution\n",
    "get_pvalue <- function(s1_mC,s1_cov, s2_mC,s2_cov,\n",
    "                       c1_mC,c1_cov,c2_mC,c2_cov,c3_mC,c3_cov){\n",
    "    \n",
    "    if(s1_cov == 0 & s2_cov == 0){return(NA)} \n",
    "    \n",
    "    else{\n",
    "\n",
    "    s1_mC = s1_mC +1 \n",
    "    s1_cov =  s1_cov +1 \n",
    "    s2_mC = s2_mC +1 \n",
    "    s2_cov = s2_cov +1 \n",
    "    c1_mC = c1_mC +1 \n",
    "    c1_cov = c1_cov +1 \n",
    "    c2_mC = c2_mC +1 \n",
    "    c2_cov = c2_cov +1 \n",
    "    c3_mC = c3_mC +1 \n",
    "    c3_cov = c3_cov +1 \n",
    "    \n",
    "    \n",
    "    treatment <- c('sample','sample','control','control','control')\n",
    "    success <- c(s1_mC,s2_mC,c1_mC,c2_mC,c3_mC)\n",
    "    failure <- c(s1_cov-s1_mC,s2_cov-s2_mC,c1_cov-c1_mC,c2_cov-c2_mC,c3_cov-c3_mC)\n",
    "    da <- data.frame(treatment, success,failure)\n",
    "\n",
    "    fit <- glm(cbind(da$success,da$failure) ~ treatment, family = binomial, data = da)\n",
    "    p_val <- summary(fit)$coefficients[8]\n",
    "    \n",
    "    return(p_val) }\n",
    "}\n",
    "\n",
    "get_pvalue_1rep <- function(s1_mC,s1_cov,\n",
    "                       c1_mC,c1_cov,c2_mC,c2_cov,c3_mC,c3_cov){\n",
    "    \n",
    "    if(s1_cov == 0){return(NA)}\n",
    "    \n",
    "    else{\n",
    "\n",
    "    s1_mC = s1_mC +1 \n",
    "    s1_cov =  s1_cov +1 \n",
    "    c1_mC = c1_mC +1 \n",
    "    c1_cov = c1_cov +1 \n",
    "    c2_mC = c2_mC +1 \n",
    "    c2_cov = c2_cov +1 \n",
    "    c3_mC = c3_mC +1 \n",
    "    c3_cov = c3_cov +1 \n",
    "    \n",
    "    treatment <- c('sample','control','control','control')\n",
    "    success <- c(s1_mC,c1_mC,c2_mC,c3_mC)\n",
    "    failure <- c(s1_cov-s1_mC,c1_cov-c1_mC,c2_cov-c2_mC,c3_cov-c3_mC)\n",
    "    da <- data.frame(treatment, success,failure) \n",
    "    \n",
    "    fit <- glm(cbind(da$success,da$failure) ~ treatment, family = binomial, data = da)\n",
    "    p_val <- summary(fit)$coefficients[8]\n",
    "    return(p_val) }\n",
    "}\n",
    "\n",
    "CompareCGmap <- function(strain_cgmap, TS559_cgmap, strain, minimum_freq=.10, min_cov=47){\n",
    "    \n",
    "    meth_freq_r1 <- paste(\"methylation_frequency_\", strain, \"_Dec2016\", sep=\"\")\n",
    "    m5C_count_r1 <- paste(\"mC_count_\", strain, \"_Dec2016\", sep=\"\")\n",
    "    cov_r1 <- paste(\"coverage_\", strain, \"_Dec2016\", sep=\"\")\n",
    "\n",
    "    meth_freq_r2 <- paste(\"methylation_frequency_\", strain, \"_Oct2020\", sep=\"\")\n",
    "    m5C_count_r2 <- paste(\"mC_count_\", strain, \"_Oct2020\", sep=\"\")\n",
    "    cov_r2 <- paste(\"coverage_\", strain, \"_Oct2020\", sep=\"\")\n",
    "    \n",
    "    \n",
    "    #get quantile values for m5C count\n",
    "    q_TS559_march <- quantile((TS559_cgmap%>%\n",
    "        filter(coverage_TS559_March2016>=min_cov, mC_count_TS559_March2016>=0))$mC_count_TS559_March2016, .99)[[1]]\n",
    "    \n",
    "    q_TS559_dec <- quantile((TS559_cgmap%>%\n",
    "        filter(coverage_TS559_Dec2016>=min_cov, mC_count_TS559_Dec2016>=0))$mC_count_TS559_Dec2016, .99)[[1]]\n",
    "    \n",
    "    q_TS559_oct <- quantile((TS559_cgmap%>%\n",
    "        filter(coverage_TS559_Oct2020>=min_cov, mC_count_TS559_Oct2020>=0))$mC_count_TS559_Oct2020, .99)[[1]]\n",
    "\n",
    "    q_strain_dec <- quantile(\n",
    "        (strain_cgmap %>%\n",
    "        filter(!!as.name(cov_r1)>=min_cov, !!as.name(m5C_count_r1)>=0) )[[m5C_count_r1]], .99)[[1]]\n",
    "    \n",
    "    q_strain_oct <- quantile(\n",
    "        (strain_cgmap %>%\n",
    "        filter(!!as.name(cov_r2)>=min_cov, !!as.name(m5C_count_r2)>=0) )[[m5C_count_r2]], .99)[[1]]\n",
    "    \n",
    "    \n",
    "    pooled_methylation_frequency_strain <- paste(\"pooled_methylation_frequency_\", strain, sep='')\n",
    "    \n",
    "    data <- strain_cgmap %>%\n",
    "        merge(y = TS559_cgmap, all = TRUE, \n",
    "             by= c(\"chromosome\",\"position\",\"reference_nucleotide\",\"growth_phase\",\"metabolic_condition\")) %>%\n",
    "        dplyr::select(-chromosome,-standard_devition) %>%\n",
    "    \n",
    "        # house keeping\n",
    "        mutate(!!as.name(cov_r1) := replace_na(!!as.name(cov_r1),0)) %>%\n",
    "        mutate(!!as.name(cov_r2) := replace_na(!!as.name(cov_r2),0)) %>%\n",
    "        \n",
    "        mutate(coverage_TS559_March2016 = replace_na(coverage_TS559_March2016,0)) %>%\n",
    "        mutate(coverage_TS559_Dec2016 = replace_na(coverage_TS559_Dec2016,0)) %>%\n",
    "        mutate(coverage_TS559_Oct2020 = replace_na(coverage_TS559_Oct2020,0)) %>%\n",
    "        \n",
    "        mutate(methylation_frequency_TS559_March2016 = na_if(methylation_frequency_TS559_March2016, \"NA\")) %>%    \n",
    "        mutate(methylation_frequency_TS559_Dec2016 = na_if(methylation_frequency_TS559_Dec2016, \"NA\")) %>%    \n",
    "        mutate(methylation_frequency_TS559_Oct2020 = na_if(methylation_frequency_TS559_Oct2020, \"NA\")) %>%    \n",
    "        mutate(!!as.name(meth_freq_r1) := na_if(!!as.name(meth_freq_r1), \"NA\")) %>%  \n",
    "        mutate(!!as.name(meth_freq_r2) := na_if(!!as.name(meth_freq_r2), \"NA\")) %>%    \n",
    "        \n",
    "        # identify high confidence and reproducible m5C sites\n",
    "        rowwise() %>%\n",
    "        mutate(hiconf_TS559 = detect_hiconf_3reps(\n",
    "            f1=methylation_frequency_TS559_March2016,\n",
    "            m1=mC_count_TS559_March2016,\n",
    "            p1=q_TS559_march,\n",
    "            c1=coverage_TS559_March2016,\n",
    "            f2=methylation_frequency_TS559_Dec2016,\n",
    "            m2=mC_count_TS559_Dec2016,\n",
    "            p2=q_TS559_dec,\n",
    "            c2=coverage_TS559_Dec2016,\n",
    "            f3=methylation_frequency_TS559_Oct2020,\n",
    "            m3=mC_count_TS559_Oct2020,\n",
    "            p3=q_TS559_oct,\n",
    "            c3=coverage_TS559_Oct2020,\n",
    "            min_cov=min_cov,\n",
    "            min_freq = minimum_freq) \n",
    "              ) %>%\n",
    "        \n",
    "        rowwise() %>%\n",
    "        mutate(hiconf_strain = detect_hiconf_2reps(\n",
    "                f1=!!as.name(meth_freq_r1),\n",
    "                m1=!!as.name(m5C_count_r1),\n",
    "                p1=q_strain_dec,\n",
    "                c1=!!as.name(cov_r1),\n",
    "                f2=!!as.name(meth_freq_r2),\n",
    "                m2=!!as.name(m5C_count_r2),\n",
    "                p2=q_strain_oct,\n",
    "                c2=!!as.name(cov_r2), \n",
    "                min_cov= min_cov,\n",
    "                min_freq = minimum_freq)\n",
    "                 \n",
    "                ) %>%\n",
    "        # we are only interested in keeping sites were either the parent or deletion\n",
    "        # strain have a m5C site\n",
    "        filter(hiconf_TS559 == TRUE | hiconf_strain == TRUE) %>%\n",
    "        filter(hiconf_TS559 != \"NA\") %>%\n",
    "        filter(hiconf_strain != \"NA\") %>%\n",
    "    \n",
    "        # calculate a mean methylation frequency at each site by pooling replicates (round to 2 decimals)\n",
    "        # calculate standard deviation of each site as well\n",
    "        mutate(!!as.name(pooled_methylation_frequency_strain) := \n",
    "             ( !!as.name(m5C_count_r1) +  !!as.name(m5C_count_r2) ) / ( !!as.name(cov_r1) + !!as.name(cov_r2) ) ) %>%\n",
    "        \n",
    "        mutate(!!as.name(pooled_methylation_frequency_strain) := round(!!as.name(pooled_methylation_frequency_strain),2))%>%\n",
    "        mutate(strain_sd = sd(c(!!as.name(meth_freq_r1),!!as.name(meth_freq_r2))))%>%\n",
    "    \n",
    "        mutate(pooled_methylation_frequency_TS559 =\n",
    "               (mC_count_TS559_March2016 + mC_count_TS559_Dec2016 + mC_count_TS559_Oct2020) /\n",
    "               (coverage_TS559_March2016+coverage_TS559_Dec2016+coverage_TS559_Oct2020)    ) %>%\n",
    "        mutate(pooled_methylation_frequency_TS559=round(pooled_methylation_frequency_TS559,2)) %>%\n",
    "        mutate(TS559_sd = sd(c(methylation_frequency_TS559_March2016,methylation_frequency_TS559_Dec2016,\n",
    "                              methylation_frequency_TS559_Oct2020))) %>%\n",
    "    \n",
    "\n",
    "        # calculate fold change of methylaton frequency at each site between parent and deletion strain\n",
    "        mutate(FoldChange = (!!as.name(pooled_methylation_frequency_strain)+.01) / \n",
    "                               (pooled_methylation_frequency_TS559+.01),\n",
    "               log2FC = log2( (!!as.name(pooled_methylation_frequency_strain)+.01) / \n",
    "                               (pooled_methylation_frequency_TS559+.01) ) ) %>%\n",
    "        \n",
    "        filter(log2FC >= 1 | log2FC <= -1) %>%\n",
    "        # add p vlaue\n",
    "        rowwise() %>%\n",
    "        mutate(pvalue = get_pvalue(\n",
    "            s1_mC = !!as.name(m5C_count_r1),\n",
    "            s1_cov = !!as.name(cov_r1), \n",
    "            s2_mC = !!as.name(m5C_count_r2),\n",
    "            s2_cov = !!as.name(cov_r2),\n",
    "            c1_mC = mC_count_TS559_March2016,\n",
    "            c1_cov = coverage_TS559_March2016,\n",
    "            c2_mC = mC_count_TS559_Dec2016,\n",
    "            c2_cov = coverage_TS559_Dec2016,\n",
    "            c3_mC = mC_count_TS559_Oct2020,\n",
    "            c3_cov = coverage_TS559_Oct2020) ) %>%\n",
    "        \n",
    "        filter(pvalue <= 0.01)\n",
    "        \n",
    "        # adjust p values\n",
    "        #adj_pvalue <- p.adjust(data$pvalue, method = 'fdr')\n",
    "        #data <- cbind(data, adj_pvalue)\n",
    "        data <- data %>% merge(y=annotation,by='position', all.x=TRUE) %>%\n",
    "            dplyr::rename(TS559_position = position) %>%\n",
    "            rowwise()%>%\n",
    "            mutate(overlap = \n",
    "                   rm_overlap(!!as.name(pooled_methylation_frequency_strain), strain_sd,\n",
    "                    pooled_methylation_frequency_TS559, TS559_sd )) %>%\n",
    "           filter(overlap==FALSE) %>%\n",
    "           dplyr::select(-overlap, -reproducible, -highly_reproducible) %>%\n",
    "           arrange(pvalue)\n",
    "    \n",
    "    return(data)\n",
    "}\n",
    "defaultW <- getOption(\"warn\")\n",
    "\n",
    "\n",
    "\n",
    "enumerate_hiconf <- function(strain_cgmap, strain, min_cov=47){\n",
    "    meth_freq_r1 <- paste(\"methylation_frequency_\", strain, \"_Dec2016\", sep=\"\")\n",
    "    m5C_count_r1 <- paste(\"mC_count_\", strain, \"_Dec2016\", sep=\"\")\n",
    "    cov_r1 <- paste(\"coverage_\", strain, \"_Dec2016\", sep=\"\")\n",
    "\n",
    "    meth_freq_r2 <- paste(\"methylation_frequency_\", strain, \"_Oct2020\", sep=\"\")\n",
    "    m5C_count_r2 <- paste(\"mC_count_\", strain, \"_Oct2020\", sep=\"\")\n",
    "    cov_r2 <- paste(\"coverage_\", strain, \"_Oct2020\", sep=\"\")\n",
    "    \n",
    "    q_strain_dec <- quantile(\n",
    "        (strain_cgmap %>%\n",
    "        filter(!!as.name(cov_r1)>=min_cov, !!as.name(m5C_count_r1)>=0) )[[m5C_count_r1]], .99)[[1]]\n",
    "    \n",
    "    q_strain_oct <- quantile(\n",
    "        (strain_cgmap %>%\n",
    "        filter(!!as.name(cov_r2)>=min_cov, !!as.name(m5C_count_r2)>=0) )[[m5C_count_r2]], .99)[[1]]\n",
    "    \n",
    "    \n",
    "    data <- strain_cgmap %>%\n",
    "        dplyr::select(-chromosome, -reference_nucleotide,-growth_phase) %>%\n",
    "    \n",
    "        mutate(!!as.name(cov_r1) := replace_na(!!as.name(cov_r1),0)) %>%\n",
    "        mutate(!!as.name(cov_r2) := replace_na(!!as.name(cov_r2),0)) %>%\n",
    "        \n",
    "        mutate(!!as.name(meth_freq_r1) := na_if(!!as.name(meth_freq_r1), \"NA\")) %>%  \n",
    "        mutate(!!as.name(meth_freq_r2) := na_if(!!as.name(meth_freq_r2), \"NA\")) %>%    \n",
    "        \n",
    "        rowwise() %>%\n",
    "        mutate(hiconf_strain = detect_hiconf_2reps(\n",
    "                f1=!!as.name(meth_freq_r1),\n",
    "                m1=!!as.name(m5C_count_r1),\n",
    "                p1=q_strain_dec,\n",
    "                c1=!!as.name(cov_r1),\n",
    "                f2=!!as.name(meth_freq_r2),\n",
    "                m2=!!as.name(m5C_count_r2),\n",
    "                p2=q_strain_oct,\n",
    "                c2=!!as.name(cov_r2), \n",
    "                min_cov=min_cov)\n",
    "                 \n",
    "                ) %>%\n",
    "        filter(hiconf_strain == TRUE) %>%\n",
    "        filter(!!as.name(cov_r1) >= min_cov, !!as.name(cov_r2) >= min_cov) %>%\n",
    "\n",
    "        dplyr::rename(TS559_position = position) %>%\n",
    "        \n",
    "        mutate( 'm5C_cov/total_cov_Dec2016' := \n",
    "            paste(!!as.name(paste('mC_count_',strain,'_Dec2016',sep='')), \n",
    "                  '/', \n",
    "                  !!as.name(paste('coverage_',strain,'_Dec2016',sep='')) \n",
    "                 ) \n",
    "              ) %>%\n",
    "        \n",
    "        mutate( 'm5C_cov/total_cov_Oct2020' := \n",
    "            paste(!!as.name(paste('mC_count_',strain,'_Oct2020',sep='')), \n",
    "                  '/', \n",
    "                  !!as.name(paste('coverage_',strain,'_Oct2020',sep='')) \n",
    "                 ) \n",
    "              ) %>%\n",
    "        dplyr::select(TS559_position, \n",
    "                  !!as.name(meth_freq_r1), \n",
    "                  'm5C_cov/total_cov_Dec2016',\n",
    "                  !!as.name(meth_freq_r2), \n",
    "                  'm5C_cov/total_cov_Oct2020',\n",
    "                     hiconf_strain) %>%\n",
    "\n",
    "    dplyr::rename(!!paste(strain,'_hiconf',sep=''):= hiconf_strain )  %>%\n",
    "    dplyr::rename(!!paste('m5C_cov/total_cov_',strain,'_rep1',sep=''):= 'm5C_cov/total_cov_Dec2016' )  %>%\n",
    "    dplyr::rename(!!paste('m5C_cov/total_cov_',strain,'_rep2',sep=''):= 'm5C_cov/total_cov_Oct2020' )\n",
    "    \n",
    "    return(data)\n",
    "}\n",
    "\n",
    "get_quantiles <- function(strain_cgmap, strain, min_cov = 47){\n",
    "    \n",
    "    m5C_count_r1 <- paste(\"mC_count_\", strain, \"_Dec2016\", sep=\"\")\n",
    "    cov_r1 <- paste(\"coverage_\", strain, \"_Dec2016\", sep=\"\")\n",
    "\n",
    "    m5C_count_r2 <- paste(\"mC_count_\", strain, \"_Oct2020\", sep=\"\")\n",
    "    cov_r2 <- paste(\"coverage_\", strain, \"_Oct2020\", sep=\"\")\n",
    "    \n",
    "    \n",
    "    #get quantile values for m5C count\n",
    "    q_strain_dec <- quantile(\n",
    "        (strain_cgmap %>%\n",
    "        filter(!!as.name(cov_r1)>=min_cov, !!as.name(m5C_count_r1)>=0) )[[m5C_count_r1]], .99)[[1]]\n",
    "    \n",
    "    q_strain_oct <- quantile(\n",
    "        (strain_cgmap %>%\n",
    "        filter(!!as.name(cov_r2)>=min_cov, !!as.name(m5C_count_r2)>=0) )[[m5C_count_r2]], .99)[[1]]\n",
    "\n",
    "    return(c(q_strain_dec,q_strain_oct))\n",
    "}\n",
    "\n",
    "defaultW <- getOption(\"warn\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "###strain TK0008\n",
    "\n",
    "CompareCGmap_1rep <- function(strain_cgmap, TS559_cgmap, strain, minimum_freq=.10, min_cov=47){\n",
    "    \n",
    "    meth_freq_r1 <- paste(\"methylation_frequency_\", strain, \"_Dec2016\", sep=\"\")\n",
    "    m5C_count_r1 <- paste(\"mC_count_\", strain, \"_Dec2016\", sep=\"\")\n",
    "    cov_r1 <- paste(\"coverage_\", strain, \"_Dec2016\", sep=\"\")\n",
    "\n",
    "    meth_freq_r2 <- paste(\"methylation_frequency_\", strain, \"_Oct2020\", sep=\"\")\n",
    "    m5C_count_r2 <- paste(\"mC_count_\", strain, \"_Oct2020\", sep=\"\")\n",
    "    cov_r2 <- paste(\"coverage_\", strain, \"_Oct2020\", sep=\"\")\n",
    "    \n",
    "    \n",
    "    #get quantile values for m5C count\n",
    "    q_TS559_march <- quantile((TS559_cgmap%>%\n",
    "        filter(coverage_TS559_March2016>=min_cov, mC_count_TS559_March2016>=0))$mC_count_TS559_March2016, .99)[[1]]\n",
    "    \n",
    "    q_TS559_dec <- quantile((TS559_cgmap%>%\n",
    "        filter(coverage_TS559_Dec2016>=min_cov, mC_count_TS559_Dec2016>=0))$mC_count_TS559_Dec2016, .99)[[1]]\n",
    "    \n",
    "    q_TS559_oct <- quantile((TS559_cgmap%>%\n",
    "        filter(coverage_TS559_Oct2020>=min_cov, mC_count_TS559_Oct2020>=0))$mC_count_TS559_Oct2020, .99)[[1]]\n",
    "\n",
    "    q_strain_dec <- quantile(\n",
    "        (strain_cgmap %>%\n",
    "        filter(!!as.name(cov_r1)>=min_cov, !!as.name(m5C_count_r1)>=0) )[[m5C_count_r1]], .99)[[1]]\n",
    "\n",
    "    \n",
    "    data <- strain_cgmap %>%\n",
    "        merge(y = TS559_cgmap, all = TRUE, \n",
    "             by= c(\"chromosome\",\"position\",\"reference_nucleotide\",\"growth_phase\",\"metabolic_condition\")) %>%\n",
    "        dplyr::select(-chromosome,-standard_devition) %>%\n",
    "    \n",
    "        # house keeping\n",
    "        mutate(!!as.name(cov_r1) := replace_na(!!as.name(cov_r1),0)) %>%\n",
    "        mutate(!!as.name(cov_r2) := replace_na(!!as.name(cov_r2),0)) %>%\n",
    "        \n",
    "        mutate(coverage_TS559_March2016 = replace_na(coverage_TS559_March2016,0)) %>%\n",
    "        mutate(coverage_TS559_Dec2016 = replace_na(coverage_TS559_Dec2016,0)) %>%\n",
    "        mutate(coverage_TS559_Oct2020 = replace_na(coverage_TS559_Oct2020,0)) %>%\n",
    "        \n",
    "        mutate(methylation_frequency_TS559_March2016 = na_if(methylation_frequency_TS559_March2016, \"NA\")) %>%    \n",
    "        mutate(methylation_frequency_TS559_Dec2016 = na_if(methylation_frequency_TS559_Dec2016, \"NA\")) %>%    \n",
    "        mutate(methylation_frequency_TS559_Oct2020 = na_if(methylation_frequency_TS559_Oct2020, \"NA\")) %>%    \n",
    "        \n",
    "        mutate(!!as.name(meth_freq_r1) := na_if(!!as.name(meth_freq_r1), \"NA\")) %>%  \n",
    "        \n",
    "        dplyr::select(-!!as.name(meth_freq_r2),\n",
    "                     -!!as.name(m5C_count_r2),\n",
    "                     -!!as.name(cov_r2)) %>%\n",
    "    \n",
    "        # identify high confidence and reproducible m5C sites\n",
    "        rowwise() %>%\n",
    "        mutate(hiconf_TS559 = detect_hiconf_3reps(\n",
    "            f1=methylation_frequency_TS559_March2016,\n",
    "            m1=mC_count_TS559_March2016,\n",
    "            p1=q_TS559_march,\n",
    "            c1=coverage_TS559_March2016,\n",
    "            f2=methylation_frequency_TS559_Dec2016,\n",
    "            m2=mC_count_TS559_Dec2016,\n",
    "            p2=q_TS559_dec,\n",
    "            c2=coverage_TS559_Dec2016,\n",
    "            f3=methylation_frequency_TS559_Oct2020,\n",
    "            m3=mC_count_TS559_Oct2020,\n",
    "            p3=q_TS559_oct,\n",
    "            c3=coverage_TS559_Oct2020,\n",
    "            min_cov=min_cov,\n",
    "            min_freq = minimum_freq) ) %>%\n",
    "        \n",
    "        rowwise() %>%\n",
    "    \n",
    "    mutate(hiconf_strain = detect_hiconf_1rep(\n",
    "                f1=!!as.name(meth_freq_r1),\n",
    "                m1=!!as.name(m5C_count_r1),\n",
    "                p1=q_strain_dec,\n",
    "                c1=!!as.name(cov_r1), \n",
    "                min_cov=min_cov) ) %>%\n",
    "           \n",
    "           \n",
    "        # we are only interested in keeping sites were either the parent or deletion\n",
    "        # strain have a m5C site\n",
    "        filter(hiconf_TS559 == TRUE | hiconf_strain == TRUE) %>%\n",
    "        mutate(hiconf_TS559 = na_if(hiconf_TS559, \"NA\")) %>%    \n",
    "        mutate(hiconf_TS559 = na_if(hiconf_TS559, \"NA\")) %>%    \n",
    "        filter(hiconf_TS559 != \"NA\" & hiconf_strain != \"NA\") %>%\n",
    "    \n",
    "        # calculate a mean methylation frequency at each site by pooling replicates (round to 2 decimals)\n",
    "    \n",
    "        mutate(pooled_methylation_frequency_TS559 =\n",
    "               (mC_count_TS559_March2016 + mC_count_TS559_Dec2016 + mC_count_TS559_Oct2020) /\n",
    "               (coverage_TS559_March2016+coverage_TS559_Dec2016+coverage_TS559_Oct2020)    ) %>%\n",
    "        mutate(pooled_methylation_frequency_TS559=round(pooled_methylation_frequency_TS559,2)) %>%\n",
    "    \n",
    "        # caclutate standard deviation of each site as well\n",
    "        mutate(strain_sd = 0)%>%\n",
    "\n",
    "        mutate(TS559_sd = sd(c(methylation_frequency_TS559_March2016,methylation_frequency_TS559_Dec2016,\n",
    "                              methylation_frequency_TS559_Oct2020))) %>%\n",
    "        \n",
    "        \n",
    "        # calcualte fold change of methylaton frequency at each site between parent and deletion strain\n",
    "        mutate(FoldChange = (!!as.name(meth_freq_r1) / pooled_methylation_frequency_TS559 ),\n",
    "            log2FC = log2(!!as.name(meth_freq_r1) / pooled_methylation_frequency_TS559 ) ) %>%\n",
    "        \n",
    "        filter(log2FC >= 1 | log2FC <= -1) %>%\n",
    "\n",
    "        # add p vlaue\n",
    "        rowwise() %>%\n",
    "        mutate(pvalue = get_pvalue_1rep(\n",
    "            s1_mC = !!as.name(m5C_count_r1),\n",
    "            s1_cov = !!as.name(cov_r1), \n",
    "            c1_mC = mC_count_TS559_March2016,\n",
    "            c1_cov = coverage_TS559_March2016,\n",
    "            c2_mC = mC_count_TS559_Dec2016,\n",
    "            c2_cov = coverage_TS559_Dec2016,\n",
    "            c3_mC = mC_count_TS559_Oct2020,\n",
    "            c3_cov = coverage_TS559_Oct2020) ) %>%\n",
    "        \n",
    "        filter(pvalue <= 0.05)\n",
    "        \n",
    "        data <- data %>% merge(y=annotation,by='position', all.x=TRUE) %>%\n",
    "            dplyr::rename(TS559_position = position) %>%\n",
    "            rowwise()%>%\n",
    "            mutate(overlap = \n",
    "                   rm_overlap(!!as.name(meth_freq_r1), strain_sd,\n",
    "                    pooled_methylation_frequency_TS559, TS559_sd )) %>%\n",
    "           filter(overlap==FALSE) %>%\n",
    "           dplyr::select(-reproducible, -highly_reproducible) %>%\n",
    "           arrange(log2FC)\n",
    "    return(data)\n",
    "}\n",
    "\n",
    "get_quantiles_1rep <- function(strain_cgmap, strain, min_cov = 47){\n",
    "    \n",
    "    m5C_count_r1 <- paste(\"mC_count_\", strain, \"_Dec2016\", sep=\"\")\n",
    "    cov_r1 <- paste(\"coverage_\", strain, \"_Dec2016\", sep=\"\")\n",
    "\n",
    "    \n",
    "    #get quantile values for m5C count\n",
    "    q_strain_dec <- quantile(\n",
    "        (strain_cgmap %>%\n",
    "        filter(!!as.name(cov_r1)>=min_cov, !!as.name(m5C_count_r1)>=0) )[[m5C_count_r1]], .99)[[1]]\n",
    "\n",
    "    return(c(q_strain_dec))\n",
    "}\n",
    "\n",
    "defaultW <- getOption(\"warn\")\n",
    "options(warn = -1)\n",
    "\n",
    "\n",
    "#load CGmaps into session\n",
    "TK0008exoS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK0008exoS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0008\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK0008statS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK0008statS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0008\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK0008exoS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK0008exoS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0008\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "TK0008statS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK0008statS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0008\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "# process raw CGmaps\n",
    "TK0008exoS_Dec2016_df <- processCGmap(TK0008exoS_CGmap_Dec2016)\n",
    "TK0008exoS_Oct2020_df <- processCGmap(TK0008exoS_CGmap_Oct2020)\n",
    "TK0008statS_Dec2016_df <- processCGmap(TK0008statS_CGmap_Dec2016)\n",
    "TK0008statS_Oct2020_df <- processCGmap(TK0008statS_CGmap_Oct2020)\n",
    "\n",
    "#merge reps\n",
    "TK0008exoS <- merge_del_reps(x=TK0008exoS_Dec2016_df, y=TK0008exoS_Oct2020_df, strain='TK0008')\n",
    "TK0008statS <- merge_del_reps(x=TK0008statS_Dec2016_df, y=TK0008statS_Oct2020_df,  strain='TK0008')\n",
    "\n",
    "\n",
    "\n",
    "eTK0008exoS <- enumerate_hiconf(TK0008exoS, \"TK0008\")\n",
    "eTK0008statS <- enumerate_hiconf(TK0008statS, \"TK0008\")\n",
    "\n",
    "\n",
    "options(warn = defaultW)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "###strain TK0224\n",
    "\n",
    "#load CGmaps into session\n",
    "TK0224exoS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK0224exoS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0224\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK0224statS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK0224statS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0224\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK0224exoS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK0224exoS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0224\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "TK0224statS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK0224statS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0224\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "\n",
    "# process raw CGmaps\n",
    "TK0224exoS_Dec2016_df <- processCGmap(TK0224exoS_CGmap_Dec2016)\n",
    "TK0224exoS_Oct2020_df <- processCGmap(TK0224exoS_CGmap_Oct2020)\n",
    "TK0224statS_Dec2016_df <- processCGmap(TK0224statS_CGmap_Dec2016)\n",
    "TK0224statS_Oct2020_df <- processCGmap(TK0224statS_CGmap_Oct2020)\n",
    "\n",
    "#merge reps\n",
    "TK0224exoS <- merge_del_reps(x=TK0224exoS_Dec2016_df, y=TK0224exoS_Oct2020_df, strain='TK0224')\n",
    "TK0224statS <- merge_del_reps(x=TK0224statS_Dec2016_df, y=TK0224statS_Oct2020_df,  strain='TK0224')\n",
    "\n",
    "\n",
    "eTK0224exoS <- enumerate_hiconf(TK0224exoS, \"TK0224\")\n",
    "eTK0224statS <- enumerate_hiconf(TK0224statS, \"TK0224\")\n",
    "\n",
    "\n",
    "options(warn = defaultW)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "###strain TK0234\n",
    "\n",
    "#load CGmaps into session\n",
    "TK0234exoS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK0234exoS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0234\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK0234statS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK0234statS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0234\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK0234exoS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK0234exoS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0234\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "TK0234statS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK0234statS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0234\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "\n",
    "# process raw CGmaps\n",
    "TK0234exoS_Dec2016_df <- processCGmap(TK0234exoS_CGmap_Dec2016)\n",
    "TK0234exoS_Oct2020_df <- processCGmap(TK0234exoS_CGmap_Oct2020)\n",
    "TK0234statS_Dec2016_df <- processCGmap(TK0234statS_CGmap_Dec2016)\n",
    "TK0234statS_Oct2020_df <- processCGmap(TK0234statS_CGmap_Oct2020)\n",
    "\n",
    "#merge reps\n",
    "TK0234exoS <- merge_del_reps(x=TK0234exoS_Dec2016_df, y=TK0234exoS_Oct2020_df, strain='TK0234')\n",
    "TK0234statS <- merge_del_reps(x=TK0234statS_Dec2016_df, y=TK0234statS_Oct2020_df,  strain='TK0234')\n",
    "\n",
    "\n",
    "eTK0234exoS <- enumerate_hiconf(TK0234exoS, \"TK0234\")\n",
    "eTK0234statS <- enumerate_hiconf(TK0234statS, \"TK0234\")\n",
    "\n",
    "\n",
    "options(warn = defaultW)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "###strain TK0704\n",
    "\n",
    "#load CGmaps into session\n",
    "TK0704exoS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK0704exoS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0704\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK0704statS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK0704statS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0704\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK0704exoS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK0704exoS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0704\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "TK0704statS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK0704statS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0704\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "\n",
    "# process raw CGmaps\n",
    "TK0704exoS_Dec2016_df <- processCGmap(TK0704exoS_CGmap_Dec2016)\n",
    "TK0704exoS_Oct2020_df <- processCGmap(TK0704exoS_CGmap_Oct2020)\n",
    "TK0704statS_Dec2016_df <- processCGmap(TK0704statS_CGmap_Dec2016)\n",
    "TK0704statS_Oct2020_df <- processCGmap(TK0704statS_CGmap_Oct2020)\n",
    "\n",
    "#merge reps\n",
    "TK0704exoS <- merge_del_reps(x=TK0704exoS_Dec2016_df, y=TK0704exoS_Oct2020_df, strain='TK0704')\n",
    "TK0704statS <- merge_del_reps(x=TK0704statS_Dec2016_df, y=TK0704statS_Oct2020_df,  strain='TK0704')\n",
    "\n",
    "eTK0704exoS <- enumerate_hiconf(TK0704exoS, \"TK0704\")\n",
    "eTK0704statS <- enumerate_hiconf(TK0704statS, \"TK0704\")\n",
    "\n",
    "\n",
    "options(warn = defaultW)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "###strain TK0729\n",
    "\n",
    "#load CGmaps into session\n",
    "TK0729exoS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK0729exoS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0729\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK0729statS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK0729statS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0729\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK0729exoS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK0729exoS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0729\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "TK0729statS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK0729statS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0729\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "\n",
    "# process raw CGmaps\n",
    "TK0729exoS_Dec2016_df <- processCGmap(TK0729exoS_CGmap_Dec2016)\n",
    "TK0729exoS_Oct2020_df <- processCGmap(TK0729exoS_CGmap_Oct2020)\n",
    "TK0729statS_Dec2016_df <- processCGmap(TK0729statS_CGmap_Dec2016)\n",
    "TK0729statS_Oct2020_df <- processCGmap(TK0729statS_CGmap_Oct2020)\n",
    "\n",
    "#merge reps\n",
    "TK0729exoS <- merge_del_reps(x=TK0729exoS_Dec2016_df, y=TK0729exoS_Oct2020_df, strain='TK0729')\n",
    "TK0729statS <- merge_del_reps(x=TK0729statS_Dec2016_df, y=TK0729statS_Oct2020_df,  strain='TK0729')\n",
    "\n",
    "\n",
    "eTK0729exoS <- enumerate_hiconf(TK0729exoS, \"TK0729\")\n",
    "eTK0729statS <- enumerate_hiconf(TK0729statS, \"TK0729\")\n",
    "\n",
    "\n",
    "options(warn = defaultW)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "###strain TK0872\n",
    "\n",
    "#load CGmaps into session\n",
    "TK0872exoS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK0872exoS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0872\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK0872statS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK0872statS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0872\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK0872exoS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK0872exoS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0872\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "TK0872statS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK0872statS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK0872\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "\n",
    "# process raw CGmaps\n",
    "TK0872exoS_Dec2016_df <- processCGmap(TK0872exoS_CGmap_Dec2016)\n",
    "TK0872exoS_Oct2020_df <- processCGmap(TK0872exoS_CGmap_Oct2020)\n",
    "TK0872statS_Dec2016_df <- processCGmap(TK0872statS_CGmap_Dec2016)\n",
    "TK0872statS_Oct2020_df <- processCGmap(TK0872statS_CGmap_Oct2020)\n",
    "\n",
    "#merge reps\n",
    "TK0872exoS <- merge_del_reps(x=TK0872exoS_Dec2016_df, y=TK0872exoS_Oct2020_df, strain='TK0872')\n",
    "TK0872statS <- merge_del_reps(x=TK0872statS_Dec2016_df, y=TK0872statS_Oct2020_df,  strain='TK0872')\n",
    "\n",
    "eTK0872exoS <- enumerate_hiconf(TK0872exoS, \"TK0872\")\n",
    "eTK0872statS <- enumerate_hiconf(TK0872statS, \"TK0872\")\n",
    "\n",
    "\n",
    "options(warn = defaultW)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "###strain TK1273\n",
    "\n",
    "#load CGmaps into session\n",
    "TK1273exoS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK1273exoS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK1273\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK1273statS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK1273statS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK1273\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK1273exoS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK1273exoS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK1273\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "TK1273statS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK1273statS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK1273\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "\n",
    "# process raw CGmaps\n",
    "TK1273exoS_Dec2016_df <- processCGmap(TK1273exoS_CGmap_Dec2016)\n",
    "TK1273exoS_Oct2020_df <- processCGmap(TK1273exoS_CGmap_Oct2020)\n",
    "TK1273statS_Dec2016_df <- processCGmap(TK1273statS_CGmap_Dec2016)\n",
    "TK1273statS_Oct2020_df <- processCGmap(TK1273statS_CGmap_Oct2020)\n",
    "\n",
    "#merge reps\n",
    "TK1273exoS <- merge_del_reps(x=TK1273exoS_Dec2016_df, y=TK1273exoS_Oct2020_df, strain='TK1273')\n",
    "TK1273statS <- merge_del_reps(x=TK1273statS_Dec2016_df, y=TK1273statS_Oct2020_df,  strain='TK1273')\n",
    "\n",
    "eTK1273exoS <- enumerate_hiconf(TK1273exoS, \"TK1273\")\n",
    "eTK1273statS <- enumerate_hiconf(TK1273statS, \"TK1273\")\n",
    "\n",
    "\n",
    "options(warn = defaultW)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "###strain TK1917\n",
    "\n",
    "#load CGmaps into session\n",
    "TK1917exoS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK1917exoS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK1917\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK1917statS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK1917statS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK1917\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK1917exoS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK1917exoS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK1917\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "TK1917statS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK1917statS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK1917\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "\n",
    "# process raw CGmaps\n",
    "TK1917exoS_Dec2016_df <- processCGmap(TK1917exoS_CGmap_Dec2016)\n",
    "TK1917exoS_Oct2020_df <- processCGmap(TK1917exoS_CGmap_Oct2020)\n",
    "TK1917statS_Dec2016_df <- processCGmap(TK1917statS_CGmap_Dec2016)\n",
    "TK1917statS_Oct2020_df <- processCGmap(TK1917statS_CGmap_Oct2020)\n",
    "\n",
    "#merge reps\n",
    "TK1917exoS <- merge_del_reps(x=TK1917exoS_Dec2016_df, y=TK1917exoS_Oct2020_df, strain='TK1917')\n",
    "TK1917statS <- merge_del_reps(x=TK1917statS_Dec2016_df, y=TK1917statS_Oct2020_df,  strain='TK1917')\n",
    "\n",
    "eTK1917exoS <- enumerate_hiconf(TK1917exoS, \"TK1917\")\n",
    "eTK1917statS <- enumerate_hiconf(TK1917statS, \"TK1917\")\n",
    "\n",
    "\n",
    "options(warn = defaultW)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "###strain TK1935\n",
    "\n",
    "#load CGmaps into session\n",
    "TK1935exoS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK1935exoS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK1935\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK1935statS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK1935statS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK1935\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK1935exoS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK1935exoS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK1935\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "TK1935statS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK1935statS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK1935\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "\n",
    "# process raw CGmaps\n",
    "TK1935exoS_Dec2016_df <- processCGmap(TK1935exoS_CGmap_Dec2016)\n",
    "TK1935exoS_Oct2020_df <- processCGmap(TK1935exoS_CGmap_Oct2020)\n",
    "TK1935statS_Dec2016_df <- processCGmap(TK1935statS_CGmap_Dec2016)\n",
    "TK1935statS_Oct2020_df <- processCGmap(TK1935statS_CGmap_Oct2020)\n",
    "\n",
    "#merge reps\n",
    "TK1935exoS <- merge_del_reps(x=TK1935exoS_Dec2016_df, y=TK1935exoS_Oct2020_df, strain='TK1935')\n",
    "TK1935statS <- merge_del_reps(x=TK1935statS_Dec2016_df, y=TK1935statS_Oct2020_df,  strain='TK1935')\n",
    "\n",
    "\n",
    "eTK1935exoS <- enumerate_hiconf(TK1935exoS, \"TK1935\")\n",
    "eTK1935statS <- enumerate_hiconf(TK1935statS, \"TK1935\")\n",
    "\n",
    "\n",
    "options(warn = defaultW)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "###strain TK2122\n",
    "\n",
    "#load CGmaps into session\n",
    "TK2122exoS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK2122exoS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK2122\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK2122statS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK2122statS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK2122\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK2122exoS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK2122exoS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK2122\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "TK2122statS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK2122statS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK2122\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "\n",
    "# process raw CGmaps\n",
    "TK2122exoS_Dec2016_df <- processCGmap(TK2122exoS_CGmap_Dec2016)\n",
    "TK2122exoS_Oct2020_df <- processCGmap(TK2122exoS_CGmap_Oct2020)\n",
    "TK2122statS_Dec2016_df <- processCGmap(TK2122statS_CGmap_Dec2016)\n",
    "TK2122statS_Oct2020_df <- processCGmap(TK2122statS_CGmap_Oct2020)\n",
    "\n",
    "#merge reps\n",
    "TK2122exoS <- merge_del_reps(x=TK2122exoS_Dec2016_df, y=TK2122exoS_Oct2020_df, strain='TK2122')\n",
    "TK2122statS <- merge_del_reps(x=TK2122statS_Dec2016_df, y=TK2122statS_Oct2020_df,  strain='TK2122')\n",
    "\n",
    "eTK2122exoS <- enumerate_hiconf(TK2122exoS, \"TK2122\")\n",
    "eTK2122statS <- enumerate_hiconf(TK2122statS, \"TK2122\")\n",
    "\n",
    "\n",
    "options(warn = defaultW)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "###strain TK2241\n",
    "\n",
    "#load CGmaps into session\n",
    "TK2241exoS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK2241exoS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK2241\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK2241statS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK2241statS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK2241\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK2241exoS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK2241exoS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK2241\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "TK2241statS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK2241statS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK2241\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "\n",
    "# process raw CGmaps\n",
    "TK2241exoS_Dec2016_df <- processCGmap(TK2241exoS_CGmap_Dec2016)\n",
    "TK2241exoS_Oct2020_df <- processCGmap(TK2241exoS_CGmap_Oct2020)\n",
    "TK2241statS_Dec2016_df <- processCGmap(TK2241statS_CGmap_Dec2016)\n",
    "TK2241statS_Oct2020_df <- processCGmap(TK2241statS_CGmap_Oct2020)\n",
    "\n",
    "#merge reps\n",
    "TK2241exoS <- merge_del_reps(x=TK2241exoS_Dec2016_df, y=TK2241exoS_Oct2020_df, strain='TK2241')\n",
    "TK2241statS <- merge_del_reps(x=TK2241statS_Dec2016_df, y=TK2241statS_Oct2020_df,  strain='TK2241')\n",
    "\n",
    "\n",
    "eTK2241exoS <- enumerate_hiconf(TK2241exoS, \"TK2241\")\n",
    "eTK2241statS <- enumerate_hiconf(TK2241statS, \"TK2241\")\n",
    "\n",
    "\n",
    "options(warn = defaultW)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "###strain TK2304\n",
    "\n",
    "#load CGmaps into session\n",
    "TK2304exoS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK2304exoS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK2304\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK2304statS_CGmap_Dec2016 <-read.delim(\"../cgmaps/TK2304statS_totalRNA_Dec2016.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK2304\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Dec2016\")\n",
    "\n",
    "TK2304exoS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK2304exoS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK2304\", growth_phase='exoponential', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "TK2304statS_CGmap_Oct2020 <-read.delim(\"../cgmaps/TK2304statS_totalRNA_Oct2020.CGmap\", sep='\\t', header=FALSE,col.names=cols) %>%\n",
    "    mutate(strain=\"TK2304\", growth_phase='stationary', metabolic_condition=\"S\", replicate=\"Oct2020\")\n",
    "\n",
    "\n",
    "# process raw CGmaps\n",
    "TK2304exoS_Dec2016_df <- processCGmap(TK2304exoS_CGmap_Dec2016)\n",
    "TK2304exoS_Oct2020_df <- processCGmap(TK2304exoS_CGmap_Oct2020)\n",
    "TK2304statS_Dec2016_df <- processCGmap(TK2304statS_CGmap_Dec2016)\n",
    "TK2304statS_Oct2020_df <- processCGmap(TK2304statS_CGmap_Oct2020)\n",
    "\n",
    "#merge reps\n",
    "TK2304exoS <- merge_del_reps(x=TK2304exoS_Dec2016_df, y=TK2304exoS_Oct2020_df, strain='TK2304')\n",
    "TK2304statS <- merge_del_reps(x=TK2304statS_Dec2016_df, y=TK2304statS_Oct2020_df,  strain='TK2304')\n",
    "\n",
    "\n",
    "eTK2304exoS <- enumerate_hiconf(TK2304exoS, \"TK2304\")\n",
    "eTK2304statS <- enumerate_hiconf(TK2304statS, \"TK2304\")\n",
    "\n",
    "\n",
    "options(warn = defaultW)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A rowwise_df: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>TS559_position</th><th scope=col>methylation_frequency_TK2304_Dec2016</th><th scope=col>m5C_cov/total_cov_TK2304_rep1</th><th scope=col>methylation_frequency_TK2304_Oct2020</th><th scope=col>m5C_cov/total_cov_TK2304_rep2</th><th scope=col>TK2304_hiconf</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td> 28324</td><td>0.18</td><td>42 / 235</td><td>0.10</td><td>136 / 1422</td><td>TRUE</td></tr>\n",
       "\t<tr><td> 32093</td><td>0.11</td><td>20 / 180</td><td>0.09</td><td>68 / 738  </td><td>TRUE</td></tr>\n",
       "\t<tr><td>109789</td><td>0.07</td><td>59 / 835</td><td>0.07</td><td>132 / 1798</td><td>TRUE</td></tr>\n",
       "\t<tr><td>122844</td><td>0.13</td><td>14 / 105</td><td>0.10</td><td>47 / 455  </td><td>TRUE</td></tr>\n",
       "\t<tr><td>152629</td><td>0.24</td><td>52 / 213</td><td>0.21</td><td>151 / 725 </td><td>TRUE</td></tr>\n",
       "\t<tr><td>155718</td><td>0.18</td><td>66 / 369</td><td>0.11</td><td>223 / 2096</td><td>TRUE</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A rowwise\\_df: 6 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " TS559\\_position & methylation\\_frequency\\_TK2304\\_Dec2016 & m5C\\_cov/total\\_cov\\_TK2304\\_rep1 & methylation\\_frequency\\_TK2304\\_Oct2020 & m5C\\_cov/total\\_cov\\_TK2304\\_rep2 & TK2304\\_hiconf\\\\\n",
       " <int> & <dbl> & <chr> & <dbl> & <chr> & <lgl>\\\\\n",
       "\\hline\n",
       "\t  28324 & 0.18 & 42 / 235 & 0.10 & 136 / 1422 & TRUE\\\\\n",
       "\t  32093 & 0.11 & 20 / 180 & 0.09 & 68 / 738   & TRUE\\\\\n",
       "\t 109789 & 0.07 & 59 / 835 & 0.07 & 132 / 1798 & TRUE\\\\\n",
       "\t 122844 & 0.13 & 14 / 105 & 0.10 & 47 / 455   & TRUE\\\\\n",
       "\t 152629 & 0.24 & 52 / 213 & 0.21 & 151 / 725  & TRUE\\\\\n",
       "\t 155718 & 0.18 & 66 / 369 & 0.11 & 223 / 2096 & TRUE\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A rowwise_df: 6 × 6\n",
       "\n",
       "| TS559_position &lt;int&gt; | methylation_frequency_TK2304_Dec2016 &lt;dbl&gt; | m5C_cov/total_cov_TK2304_rep1 &lt;chr&gt; | methylation_frequency_TK2304_Oct2020 &lt;dbl&gt; | m5C_cov/total_cov_TK2304_rep2 &lt;chr&gt; | TK2304_hiconf &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "|  28324 | 0.18 | 42 / 235 | 0.10 | 136 / 1422 | TRUE |\n",
       "|  32093 | 0.11 | 20 / 180 | 0.09 | 68 / 738   | TRUE |\n",
       "| 109789 | 0.07 | 59 / 835 | 0.07 | 132 / 1798 | TRUE |\n",
       "| 122844 | 0.13 | 14 / 105 | 0.10 | 47 / 455   | TRUE |\n",
       "| 152629 | 0.24 | 52 / 213 | 0.21 | 151 / 725  | TRUE |\n",
       "| 155718 | 0.18 | 66 / 369 | 0.11 | 223 / 2096 | TRUE |\n",
       "\n"
      ],
      "text/plain": [
       "  TS559_position methylation_frequency_TK2304_Dec2016\n",
       "1  28324         0.18                                \n",
       "2  32093         0.11                                \n",
       "3 109789         0.07                                \n",
       "4 122844         0.13                                \n",
       "5 152629         0.24                                \n",
       "6 155718         0.18                                \n",
       "  m5C_cov/total_cov_TK2304_rep1 methylation_frequency_TK2304_Oct2020\n",
       "1 42 / 235                      0.10                                \n",
       "2 20 / 180                      0.09                                \n",
       "3 59 / 835                      0.07                                \n",
       "4 14 / 105                      0.10                                \n",
       "5 52 / 213                      0.21                                \n",
       "6 66 / 369                      0.11                                \n",
       "  m5C_cov/total_cov_TK2304_rep2 TK2304_hiconf\n",
       "1 136 / 1422                    TRUE         \n",
       "2 68 / 738                      TRUE         \n",
       "3 132 / 1798                    TRUE         \n",
       "4 47 / 455                      TRUE         \n",
       "5 151 / 725                     TRUE         \n",
       "6 223 / 2096                    TRUE         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(eTK2304statS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAPAAAADwCAIAAACxN37FAAAACXBIWXMAABJ0AAASdAHeZh94\nAAAgAElEQVR4nO2de0AU5f7/P7O7LFfFCyIoCBwFQcELebR+sepRNMVSsywzM00JxdK8HFFR\nO6EYimaaJuXdTM8vb3nUvFtHrexgSRpJXrgpIqRcdrnvZb5/DA3jzOzuzDC7Ozs8r79mn3lu\ns/PeZz/zPJ/nMxiO44BAyAWFozuAQIgJEjRCViBBI2QFEjRCViBBI2QFEjRCViBB24rLly9j\nGBYWFoYmRu0JErStWL9+PQDcvn37xIkTju5LCwJD44ctyM/P79atm5eXV2Vl5ZAhQ86fP8/M\ng2GYhRrI+0JkY94msji6g1TQCG0TNm7caDQaV61a1atXrwsXLly/fl3c+lnVjFFgLWIhg+Wy\ntGzN7r4twRFio9VqW7du3aZNm6qqqh07dgDA1KlTLRexcC+Yp5j3zuqdtZyBoyqcQjNohBaf\nHTt2aLXa+Ph4T0/PiRMn+vr67tu3r6SkRJTKLVga1PtKzUkaLdSzTCycAmsGknRAghYZk8m0\nceNGpVI5a9YsAHB1dU1MTKyvr9+yZUvzKzenZsta5FInTvkrAIZ8zdnxUsTm/wEtjCNHjgDA\nyy+/TKaUlJS4urr6+vrW1dWZK2XhXpCnuN8yyzlpp5g5LWSQvmbQCC0yH3/8MQC89957ZIqv\nr+/EiRNLS0v37dsnuFrucxoWcpKPdFYrcV7QtJ2YXLt2LTo6+qmnnrp69So1/fr16717946K\nijI33WFBZ0zj1cIts6x7alVkBmbT1BTaWen/HtAILSbM4ZmgV69eQ4YMuXHjBuuENBeI/1Pi\n2NzzmdVRnEslzAqlLF8W7GnfyJvi4mK1Wu3n51dfX888e+zYMQB4/vnnWctauBe0U+ZuHK8b\nSs3JLEWmOKNs0AgtGp9++mlDQ0NiYqJarWaeHTVqVGho6IkTJ27dutWcVnC2IdaqpeEsk27N\nBwlaHOrq6jIyMlxdXRMSElgzYBg2Z84cHMc3bNjQzLZYNQ2cbQPWUsxJa2Abhnk15BDQQ6E4\nbN++ffr06R4eHj169DCXp7a2Njs729PT8969e23btqWesvpQyDxFHZXNDcDmpA9sT4HmznLs\np0RAghaHqKio3377jWPmtLS0pKQkaooAQQOHBztzqrXw8zDXltXOSAQkaISsQDY0QlYgQSNk\nBRI0QlYgQSNkBRI0QlYgQSNkBRI0QlYgQSNkBRI0QlYgQSNkBRI0QlYgQSNkBRI0QlaoHN0B\n0aiqqpo7d66je4GwOe3bt09LSzN31t6CxjCzDqvmXHI5uurq9frHjx8vXbqUmqhSqRQKhcFg\nMJlMvPqpUqmMRiNf31qlUqlUKoU1ZzKZ+JYimtPr9QL6ieO4sOYEXJ2w5hQKhUqlojVnNBo/\n+OADC6XsJ2irUQBpIia3BrGms6JSqfz8/GgpSNCsBZ1U0AaDwUopXm00B+qmNATCRqCHQoSs\nQIJGyArnnuVYunTpqVOniGNvb28fHx/WmBgqlZDLVCgE/trt3JyLi4uwgsIQdnViNWd1X7Bz\nC7pTp04RERHEsZubm1arpZnpgt/bYPnpU/SCdm5OGPZ/CQbz6qw27dyCTkxMTExMJI7Ly8sT\nEhL0ej01g0NmOYxGoz1nOQwGg4xnOWhfpoRmORAIO+BgQbecmGsI+yAVk4MWz4oaRo3jSiEC\nAfYXtAUb35xYkYgR3JHKCI1o4eTk5BQWFnbu3LlHjx7NMUSRoBEOprS0dNasWT/88APxsV+/\nfps2bQoMDBRWG5rlQDgSHMffffddUs0AcPXq1RkzZlidnjMHEjTCkfzxxx+XL1+mJWZlZdHe\nusQdZHIg7IpWq83Kyrp165abm1tERERZWRlrtgcPHgirHwkaYT++/vrrJUuWVFZWkinBwcGs\nOTt37iysCWRyIOxEdnb2vHnzqGoGgPz8fHd3d1rO6Ojofv36CWsFCRphJ/bu3VtfX89Mr62t\npb6YZsCAAVu2bFEqlcJaQSYHwk48fPjQ3KkJEyZoNJqCgoLAwMDu3bujeWiEE1BXV2fuVGBg\nYFhYWFhYWPNbkY+gTSYThmE0b3fit65SqQT4QwseJwg3S77NKRQKYYv8gj3uBXQShH6ZOI6z\n7r0AgIiIiGHDhpnbpsD3y5SPoAlBIH9ojgXt7w/NKlmlUrl161YMw/R6fUFBQVBQEHkK+UMj\nJE2fPn2YiTExMSEhIQBQUFAgSitI0Ag7MW3aNJqV7OHhQUSNEUvNgASNsBvu7u6HDh2aMmVK\nUFCQr6/viBEjjh07FhoaKqKaQU42NEIK/Pnnn1u3br1586a3t3dcXFxcXFxRUdH27dtv3brV\nvn37sWPHpqamUvMTai4rgzNnlBMmGJvfASRohGjk5uaOGjVKq9USH48cOTJq1Khvv/22pqaG\nSDl48OCcOXMWLlxIfCTUfPu2IilJ9fAh5uoK773X3D4gkwMhGgsXLiTVTHDixAlSzQQbNmy4\nceNGQUEBoeZz55QJCS4PH2IAsH696scfG5rZByRohDjU1dVduXKFS85ffvkFAEwm2LxZtXy5\nilxvad8ePD2bu2mak8lhIcIIcQpt+0NwnA7XaDQmk6mqClu+XHXlStN42rs3npra0KtXc6NA\nWR+hqQtmKOoAwhxeXl7h4eGW82g0GgBQq7tPn+5CVfOYMcaNGxvatROhG1YETQ7A5I8PaRph\njtWrV9NSevfuTR4Tam7X7oWUlJDCwkYVKZWQmGhMSjKIFaCPxywHGSJDcPg2alUWzlLzmEtH\nSJB+/fqdPn36448/zs7Obtu2bVxc3Ntvv/3rr79u2rTJw8PDy6tVWdmYY8e6kivZbdrgqamG\nvn0bP1PXvQXDb9pOsKatBuI3F2SReQohZSIjI7dt20ZN8fX1XblyZUMDpKYqTp9ucjPq1s20\nerXB37/x5oqiZhAwD93McZoL9oyoibApxNxcaSmWlKS8ebNpkBo61LRkiZ7YqiKWlAmsCJqQ\nL01hVE2L2BWEzCDU/OuvWHKyitwLi2Hw+uuGmTONhHbEVTMIXimkhZwTEebwbMH8+Oijj/77\n3/8Sx56enhiGoYDntobj1eXl5alUqoMHsY8+UpAun15ekJJiionBCOERfna8mhMh4LljQ84x\nxU1Nqa2t1el05Ec3NzcU8Nx2cP8y8/Ly9HpYu1bx9ddNv9IuXfD0dGNwMOB4o5St1uP0Ac/5\n3p7k5OTk5GTiGAU851XQdg7+BQUFFRWwdKn6l1+aRtOnn8Y/+EDfqhVuNEJQUBDtNrEizMFf\nWoJGODWE0ZyTgy1erC4paUp/4w3TzJlGABxsYDTTkJCgWYdnNOPhLBBqPnNG8eGHLmS0AldX\nWLRIP3IkAEBgoG2lTOBgQSO9OgsnT57cv39/cXHx3/72t/j4eFogmIKCApMJMjKUX36pIu9n\nhw54WpohIsIEoAgKCuJr4QjDToIWHIgfRfCXAuvXr1+7di1x/Pvvvx8/fnzLli2jR48mUgoK\nCqqrISXF5dKlpkfAqCjTqlX69u0BAIKDg+124+z9amTaC5I5RvBnFkTYjfz8fFLNJElJSUSc\njYKCgnv3sPh4NVXNo0cbN21qVLOtjWYa/EZo2sQN8h1tCfzvf/9jJmq12uzsbB8fn++/V/zr\nXy7V1Y3pKhW8955h3Dgj2F3KjR3gnhWtC7ZMWO+7RqOprq45dUqVkaEkbWNvb1i5Uv/UUyZw\nkJqBu6CpgzF1nGYujCPkREVFRXFxsUKhoD7SaTQaV1fvrVtDLlxocjYKDTWtXm3w87PH3JwF\nJDRth5AaP/3007Rp08rLy6mJGo2mvr5dQcE/CwubxDN0qCk5We/m5kgpEyBBI9iprq5OTEyk\nqXnIkCFt2w49e3aYVts4NisUEB9vmDzZiGGOVzM0U9DIqpYxP/30Ey0ArkajuXfv7xcvPmcw\nNN53Dw9Yvlw/cKAjjWYaXAVNmsvERzQ3LHsqKiqoH2NiBt29O/bBg4FkSkAAvnq1ISTEJBEp\nEwjZgkVLFLU/CKlAxqHTaDR6veeNG1MrK7uRZwcMMKWkGFq1wiWlZhCwBctG/UBIjcjIyNGj\nR5eXl1dXB2Rnv1Vf37Qne8wY44IFBqVSKmYGFSHTdtKE+ANBDv4i8s9//nPduhs//BBtNDY2\noVbD4sWmUaMgJCTURo3SEN/B3zLSEToxHU5ztFUqlYQ/tACPYZPJxLeUQqGwp4Oy4OaIH4/l\n5vLyCj7/3PXo0QFk3b6++Nq1eFiYMSCgi16vz8/PP3HiRElJSffu3ceNG8d8mRVJc65OZH9o\n2g9C+tMaIu6vaY73iD2bE1aQLKXX6+/cuVNfXx8WFubh4UGczc4ufP991Y8/Nv1p9OqFp6UZ\nfHwUgYFBJpPp8OHDCxYsIN9qtX79+oMHD5qzQIS54hD5LTj/sCKCc5IUhmeEMM6fP//MM8/E\nxsaOGjUqOjp69+7dAHD5cuHbb7tQ1Tx6tOmTTxratWvcOlVUVJSUlER9R9uDBw9mz55t//4z\nsb7rmziQjmmBEIs//vgjISGhtraW+KjT6U6ePFlc3HXnzm5VVY15VCqYM8fw0kvGoKAg8uU9\n58+fp8UUBYCrV68WFxf7+/vbq/vsoJXClsuOHTtINQOARjPw/v2hly93JUct0tmIZktQNyZT\n0Wq1TiNoNDbLj8LCQvL42WeH5uS89ueffcmU0FBTWprB359lpjkiIoJZm6enp7kXd9sTHjY0\nZh7b9Q9hO3x9fYmD/v1f+PXXd6hqHjLElJGhf/rpLqzPeYMHDx44cCAtMSkpydXV1Xa95QhX\nQSPVyo9JkyYBQK9eb167tqCqqguRiGH4228bVqzQh4ebXTRRKBQZGRlvvvlmq1atACAwMDAt\nLe2tt96yT7ctIzBYI/VhEVkjTsrf//738eP/fejQ30ymxqc9V1fDihV4TIx19wxvb+9Vq1at\nWrWqpqaGnOyTAvx8OYBt1yrStDOSm1uwZQt24EDTgp+/v3HtWnzw4EBe9UhKzcB3CxZzYEam\niDNy7VrBkiWqa9ea7t2AAaaUFGNkZBcH9koUeG+Spc1M8ypLHrMGlKGlsDaE/gqaz9mzhYsW\nqamuzq+9Zpw1yxASIjlPIwHw84dmHgM3kVkNeG6uHi4FEVzQarVffvnlzz+3P3fuWdLhxcUF\nX7jQMGqUtHyamwNvG5p5jJA+ubm5L744zsPjvcJCDUDjYOTjg6el6UeOdHozgwpaKWwRvPPO\nQhxPLyyMIlO8vQvS012GD492YK9sAe+4HDQDAMQbrQXYyjt37szMzCSO1Wo1hmGs7rOEEynf\nzpCuC7xKNac5AaWI5ixn++ab7Fu3ltfW+pIpfn5XunY9mJenUan6C2hOWD+FOXzzbY6fg7+5\nU83XNDNQOZc67969S8b18fb29vHxYb14YcuZgidwBK+eCitl+X7v25efnNy6tpbcDGoKDv4m\nL+/9hw/h2Wf/LkBkdr46WnMiO/gzw6k7dtouOTmZfBN6ZWXl/PnzkYM/CY5Daup9amQjF5fq\n8PCdN27sJD727NmTS+BxaicVCoWAcO4ScvCXOO7u7uRGCaPRCMjB/y/q6vAZMx6cPdtkjXh6\nPujRY3tm5tfEx6FDhw4dOpRXi6we99x76DQO/qKAFmhE5MEDY1zcQ6qaBw82rlnzZ0NDnlqt\nDgwMfPfddz/77DNZfufCF1ZADBWieWXROXasMDlZ/fgxaTTD668bPvwwQKHoPHbsc0QizQa4\nd+/enj178vLyOnXqNGHChB49ejim62LAe2FFWGgOq2sx5jIIWMRpyWzceP+jj9SkYezuji9f\nbpg82ZJ7xvfffz958mQi2DMAfPHFF6tXr37llVds3VUbwcPkYJpBvAwjnAI10XIGC+kIKgYD\nnpR0f/VqFanmzp3xbdusqNlgMMyePZtUMwA0NDQkJyeXlpbatLe2g7cNjeQlQcrKTC++WLx3\nb9P/bf/+prNn/ay6zv3222+0AHYAUFNT8/3334vfS7sglYdChGB+/90wfHjJL7803coxY4wH\nDvh7e1t/wmloaOCVLn0sCZoWnRFtwZIgX39d8/zzfxYXN94CtRqWLzd8+mlnlYrTTYmIiGDd\nN9WnTx8xe2lH0AjtlFRXV5tMkJR07+23H5EGcIcO+JEj7RMSArjX06pVK/JVvCTTpk3r3r27\nWF21M5ZmOZB7ndSorq5et27d/v37tVpDx46fl5Q07b6OjDR98YWfry/vEWratGkdOnT4/PPP\nc3NzAwICJk6cSOw1dFKce6WwpTFv3rzjx48DBHl4ZJSUNDkbPf+88ZNPOrPFqeTE6NGjyZcO\nOjvWbWgu2K27LZlr164dP34c4P+pVF/W1DSqGcNM06fXfvaZcDXLDGRDOw05OTkAr2HYRoOh\n0X3FxaUGx2eNHFliuWCLgqsNjXAs9fX4wYNRAP8g74mnZ3F1dQJAkbe3t0O7Ji3kY0MTi+RO\n6uCvUCgwDGOWysrK2rVr161blffuzS0t9SHTfXx+ffToHYCa7t279+zZ0z4e97Jy8CeRrGcF\n4edEOJGSKJVKDMOMRqM9/aEFNAcATI/hgwcPzp49G6CPi8vHen0rIhHDcBzPePRoOwDu6+v7\n2WefwV+us7z6CdYCnjMhvkyTyWQ3f2gAoDVn9Uqb9WpkTHoxduXhD11VVVVeXr548WKAcRiW\npNc33ialsv5f/zIZjQFFRdO7du360ksvtWnTRq/X2+fqnMIfmvcWLKb7KHIBFZFr164tXbo0\nKysLQAWwAGA8+dW6uz+qrU0MCnp/6NB4IkWAUSR7nHsLlswoLCx87bXXdDodQFuANQBNW7Lb\ntr1VXp4AoOVrXbQ00LSdhNi8ebNOpwMIc3P7iqpmgMPl5W8AaF1cXPr27Wu2PEJOsxwy4Pbt\n2wDDlMqUurrGZRKFwmAypQL8h/i4ZMmSDh06OK6DTgDvVyMzbWhkQIsCjkNZ2WiA54xG0nWu\nsqFhtpdXvo9PcJcuXaZMmfLcc885tpPSh/dDobm5DhKkbwFotaaZM8tu3x5BprRuna/VJgA8\nmjdveUJCggP75lwgk8Px5OcbJ00qzsujjgsntdoVAPWjR4+Oj493WM+cEPTSIAfz3Xf1M2aU\n6XRNkY1wfDPALgBQqVSLFi0S/Mrklgn6shzJli3Vb7xRTqrZxaUGxxMJNQOAwWAgA50hOMI7\nLgdxgFPC93McvK2umbNmEBY1QfrU1eELFmiPHKmlpN3R6+cBFJkrUlZWtnHjxszMTBcXl5iY\nmJkzZ7Zu3doOXXUumrX0zassbXqEpksLGeShYCoPH5qmTSvPyqLGlbsEkAxQTcvZv39jdNCy\nsrJhw4aRO7QzMzNPnjx56tQpT09Pe/TYeeD3WjfWhXW0WGiB8vLy27dvU0MMXr2qHz68lKJm\nHGAXwDymmpOSksjA+mlpabR4Azk5OZs2bWpO30pKSvLz82W29CgVG1p+w3BeXt748eMjIyMH\nDx4cHh6+YcMGHMf37q15+eXHjx835nF3xwEWAHwC8IQnWnh4+M6dO6mvg//xxx+ZTbAmcuHn\nn38eMmRIdHT0s88+26dPn6+++kpYPRJEitN2FiLo0XR/9OjR7Oxs4phwKRbdH7qiomLDhg0/\n/PCDyWQaMGDAvHnz2rVrZ6EU0VxdXd2UKVPu3LlDpNfW1q5Z89G33z6dmdn0KpPgYNX771dO\nnfods56pU6fGxcVRU1j7TyTydVEqKiqaNGlSeXk58bGsrGzu3Lk+Pj7Dhw+3XLA5XyaO4xL1\nh6ZiC2ODpmbLpnZmZuapU6eIY1sEPK+qqhoxYkReXh6Rcv369W+++ebbb79t27YtM39xcfHJ\nkydLSkrCwsJ0Oh2pZgAAaAOwhqrmoKCHkyffjIwc5Ovry4y7FRMTQ7uQmJiYJysEANBoNMDf\ncX779u2kmknS09NHjBjBmp+G4C+TbxHW5qzWw3vpm1mviNYCX0/UxMTE119/nTiurq5es2aN\nuAHP09PTSTUTFBUVEW9QpeU/fvz4nDlzamsbZy2e3BYVBrAOoBMlZV9BwccpKcb0dPepU6d+\n+umn1Krmz58fEhJCXkhlZWVRUdHMmTPPnTt3//59MltkZOSMGTMAgO/VMX8YAHD37l2rwc/l\nFvCc1VnUgWoGgE6dOnXq1CgUYtQR18H/p59+YqZfuXKFVtuWLVtSU1OpiZWVlX8dxiqVK4xG\nqrPRSoBjxMfa2tpdu3bt2bPn2LFjd+7c8ff3Hz9+/PDhw4mqKisrly5deuTIEeKbf+GFF+Li\n4q5fv65SqWJiYuLj49VqNfD3uG/fvj0z0cfHx2olsnLw51idYMyp2bG7B1j/zU0mU1JSUmZm\npkqleuaZZ6qrq/fv389WGgN4E2CW0dhYiVqtbWiYDXCDmqmmpiY3N3fTpk3MrU1z5849ffo0\ncYzj+H/+858RI0YcOnSomRc1fvz4ffv20RInTJjQzGolAj/nJNG1JeXdLpmZmWq2aBf37t27\ndesWcUw+kjLwBFgJMJCS8ltDwwKAP5lZKyoqmIk3b94k1Uxy6tSpmzdvRkREMPNzp3///itW\nrFi5cmV9fT2RMnbs2FmzZjWnTulgp1kOnEPcclavPS4FbcHChQt3797NTG/durVWq7VWOhBg\nHUBXSsopgBSAetbcrIHk8vPzWTPn5uY2U9AAEB8fP2LEiEuXLtXW1vbt27d3797NrFA68HNO\nas6AylqQy3Kg/YfwI0eOMNUcEhIyfvz4AwcOWBP0MyrVGoPBg/iAYTiObyLdM5j07duXNkNH\n4OPjw0wEAF9fX9Z0vgQEBLz66quiVCUpeKwUkltiZR8K7OjRo8xENze3OXPm0Oa5GYzDsI2k\nmlWqGhyfZUHNoaGhO3bsYK2zb9++4eHhtMTw8HC0BcsyUlkplBSsRi2ROGjQIDOFXAFWAiTj\nOPmV3jUYJgCwzJMQ9O/f/8yZM+ZGXJVKlZGR0bVrk93StWvXjIwMa7+olg7yh2aB+s4RGgsW\nLDhz5kxhYeGTyR0B1gI0vTwqJCQ/L28KQI25el544YWMjAzL3QgNDT1//vylS5cKCgqCgoI0\nGo2Liwu3K2i5oJ87C6yTGwBQWlo6f/58hpp7q9UbGhpa/fURj48Hf//rKSnsao6Ojk5MTBw5\nciSXnri4uAwZMoRrvxHI5GDFy8uLmVheXt6/f/8LFy48mRynUHxOqlmprI+K2puS0nnMmNHm\nnJUVCgVHNSMEgATNwuDBg5mJdXV1Ty4OqwCSAFaYTI3/cm5uj93cZkya5L98+fLdu3dPnz6d\ntfKqqirRO4wgQSbHExQXF+fl5f3jH/84ffr0lStXzGf0BkgD6E9J+aWuLkmhqEhKSiKTWGc5\nnfo9rdIHCbqR6urqRYsWHT58mPjYr1+/hQsX/v7778XFxT///POTeUPd3DbX1VE9Ig4DrAYw\n0NxvmGp2c3NLTU0VvfMIEmRyNLJs2TJSzQBw9erVdevW3blzp2fPnk9mjFUq95BqVigMACkA\nqQDsXmDh4eHk1ERgYODRo0fRRkCbIp8Rmlgkp3m7k6G2La/+PHr06MCBA7REo9GYk5OTk5Pz\nl+WgAJgB8NaTkY3m0JyNaISHh3/33XdlZWWtW7e2MIVMRDvnu0RFetzznVQVtiJGuGoJ6KdC\noRAQ15P13onsbSdxzPkoWvVdvH//vgVvXRzHATwAUjk6G1Hp2bMnjuPEhgCO/pncYd3oyasg\nr1KkKO3jPkotS/1oOb98BE181zRdEr9vq97lOTk5FusO9PDYUlPjT0mx5GxE4ufn9+abb3Jx\nbFcoFAIi45OXLEDQAjzuyeYEFBTm4A+MFxtYrQTZ0HD79m3q1ASDZ1SqfaSaMcwEsA4g2aqa\nAUCn0z169EikbiI40dIFfenSpYSEBNIzmMEkDPvkSWejdwDo3vHmqK6u/uKLL8ToJoIr8jE5\nBLBs2bIdO3aYOanu2HFDSUl/yhshSmtrZwAU8GqCugsQYQda7gh94cIF82r2bdXq/5eUUNdN\nLtfWjuerZgAgtzwi7EOLG6HPnj27fv36mzdvmt/931ut3qDTNTkbAewG2EyLBcMFDw+PiRMn\nCu4qQgAtS9CnTp2aNm2axSxjFYrFDQ1Nr1EzGpcCXLBYpImAgADSxujQocPq1au7desmvLsI\n/rQgQeM4vmzZMvPnlQEBq+7fjyXnhdzcHtfVzQS4S83UrVs3Ly+vrKws1irGjRv31ltvZWVl\neXh49OnTB0VStD8tSNBlZWUPHjwwc7JNmzaf3b/fjfL5VkXFDIBKaiaNRvPll1/iOH7o0KFz\n586dOXOGGvckICBg1qxZ7dq169ixI985V4RYtKCHQldXVzOrr6Fubl9VVDSpuXPnixUVk0g1\nYxgWHBy8fPnyvXv3KpVKlUr16quvbt269fjx44MGDfLy8mrbtu2LL754+PBh9B55h2O/EVpY\nwHMuBTni5eWl0WguXrz4ZPJApfLDujo34oNCYTCZVhUVNW2S7dOnz6ZNm0JCQpgVRkVFMSO2\nIByLnQQtOOC51YK8SE9PHzNmzF+BlhUACQDTKM5GjZGNOnfuvG3bttLSUn9//4iICPSWEyei\nBdnQABAQEHDx4sV9+/Zdv34nK2tcbm5TONCOHSsGDTqqUkX17Tv5lVdecXNzMxqNLWprsDxo\nWYIGAE9Pz7i4t/btK8/NbXqeGznSuHlzd1dXCx4dCOfAuQV98eJFMtytyWTi6A/t66twd2/8\nqFRiM2YYly3rQquZqEqYW6Ywz2Z7+kMLc1BG/tA258yZM7SA56zh7GlGsKcn7NrVYdiwh0Yj\n9vnn7QYPdmetXHBEKGE2N2qOS3NW50OdW9ATJkwgd2g3NDTs2rWLFhBbqVRiGMa0hv38YOfO\ndj4+iuBgFWsMbcJBmW9/FAoFERJc2JApoJT9mxPgDy2sOeJPknZ1Vl9x5NyCjoyMjIyMJI7L\ny8t37drF3cE/OloF5n/xgu8BMHzSuUA0hxz8qSAHfwTC0YKWX+RShGORSsBzc4xxI9sAAANl\nSURBVBkcFfAc4aTYz4a2HPDcXAYL6UyKiormzZtHTRH8Jg3BS5LCCtq5OcE4vDmrrUv3FSd8\nwXFcp9PREgkXovT09LCwML61CTCHdu/effjw4ZUrV0ZFRdmhuX//+9/79+9///33o6Oj7dDc\noUOH9uzZs3jx4qeffppvc8DfvDx27Ni2bdvmz58/cCA1egQoFArWaJoEzj3LQQXDMNagRDqd\nzs3NzT7xijAMs2dzAKDT6dRqtX2aUygUOp3OxcXFPs0plUqdTqdSqXg1h2Y5ELICCRohK2Qu\n6LCwsNjYWLsZAN26dYuNjWV9E7gt6Nq1a2xsrLn3ZYlOcHBwbGysWK/hskqXLl1iY2P9/Px4\nlZLPQyECAbIfoREtDSRohKxw+mk7O29VFNYccwqWY4uWm7NQrS2uzlxzgq+OrNZcfgH3zrkF\nbeetioKbAzFWK1k7ae5HZYurY16s5W5waVFAfyz3E5kcCCGIsgYuOAq6BZCgxQRNGTkc5zY5\npAzzn5E8FlH3DnFFZLW17N8NVpCgbQLtlls2tQVjo2qdtBsEyOQQH8feUVsj8atDghYZid9v\n2YMELSbm1GyjnWb238DGeoGS2kcnT0Hb+StGzUmnOed+KLTzVkUupWj3g8hjo+bsf3XiFhRW\nreXmkMGHkBXyNDkQLRYkaISsQIJGyAokaISsQIJGyAokaISsQIJGyAokaISsQIJGyAokaCmC\nYRi5uks9RlgFCbqlINYPQ+I/MOd2TmoJSMTZRsoipoJGaISsQCO0CJDvCaC9MMBynBSOXpfM\nlxCweqhaOEVN5NhDJhaC5kgKNELbCtqNZ+rAagYu1VJTyAOO4hPWAYmDRmjRoG5PYh1WLcQi\n4ismq0HGqMM/dZzm20OnA43QosFXCsw4B1ZlzVQhNfgQ9Vgew60A0AhtQxyyidWm+aUPErQN\nsfPfN6uNYRlnNzCYIJNDbshPo7xAgrYhtPkyC7MKTOOYFdZpuOaYDSJWJRGQyWETyEkGXhNn\n3LFQkJipMJeBnMfg2EOnA43QtoI5Q2c5fjj36BnmUqhTHFxCo1vtoTPi9POOzghHAwMhADRC\nI2QFEjRCViBBI2QFsqERsgKN0AhZgQSNkBVI0AhZgQSNkBVI0AhZgQSNkBVI0AhZgQSNkBVI\n0AhZ8X/0m3TeHm8UPAAAAABJRU5ErkJggg==",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "options(repr.plot.width = 2, repr.plot.height = 2)\n",
    "\n",
    "lin_reg <- function(df, strain, phase) {\n",
    "    x = paste(\"methylation_frequency_\", strain, \"_Dec2016\", sep='')\n",
    "    y = paste(\"methylation_frequency_\", strain, \"_Oct2020\", sep='')\n",
    "    title = paste(\"Δ\", strain, sep='')\n",
    "    if(phase == 'exoS'){pc = 'grey90'} else if(phase == 'statS'){pc='red'}\n",
    "    \n",
    "ggplot(df, aes(x=!!as.name(x), y=!!as.name(y)))+\n",
    "    geom_point() +\n",
    "    geom_smooth(method = \"glm\", color='blue3') +\n",
    "    theme_bw() +\n",
    "    scale_x_continuous(limits=c(0,1.00)) +\n",
    "    scale_y_continuous(limits=c(0,1.00)) +\n",
    "    labs(x=\"replicate 1\", y=\"replicate 2\", title = title)+\n",
    "    theme(axis.text = element_text(color='black'),\n",
    "          plot.title = element_text(hjust = 0.5),\n",
    "          panel.background = element_rect(fill = pc)\n",
    "    )\n",
    "  \n",
    "}\n",
    "\n",
    "lin_reg(eTK2304exoS, \"TK2304\", \"exoS\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
